src/basic/efivars.c

```diff
@@ -365,10 +352,42 @@ int systemd_efi_options_variable(char **line) {
                 return -EPERM;
         }
 
-        r = efi_get_variable_string(EFI_VENDOR_SYSTEMD, "SystemdOptions", line);
+        r = efi_get_variable_string(EFI_VENDOR_SYSTEMD, "SystemdOptions", &line);
         if (r == -ENOENT)
                 return -ENODATA;
+        if (r < 0)
+                return r;
 
+        r = mkdir_safe_label("/run/systemd/efivars", 0700, 0, 0, MKDIR_WARN_MODE);
+        if (r < 0)
+                return r;
+
+        return write_string_file("/run/systemd/efivars/SystemdOptions", line,
+                        WRITE_STRING_FILE_ATOMIC|WRITE_STRING_FILE_CREATE|WRITE_STRING_FILE_MODE_0600|WRITE_STRING_FILE_SYNC);
```

Comment:
i would suggest the file is made public, i.e. no 0600 mode. It's like /proc/cmdline, and that too is readable by everyone.

Reply:
Done!